sympy/combinatorics/groups.py

```diff
@@ -0,0 +1,157 @@
+class FreeGroup(Group):
+    r"""
+    Called  with  a  positive  integer rank, FreeGroup
+    returns a free group on rank generators. If optional
+    argument name is given then the  generators are printed
+    as name1, name2 etc., that is, each name is the
+    concatenation of the string name and an integer from 1
+    to range. The default for name is the string "f".
+    """
+    is_group = True
+
+    def __new__(cls, *args, **kwargs):
+        if not(len(args) == 1 or isinstance(args[0], int)):
```

Comment:
Also I think there is some `isint` itukityy function somewhere that you should use instead of isinstance.